web/src/utils/transformDataHelper.tsx

```diff
@@ -73,14 +73,19 @@ export function sortProfileByDatetime(profileData: any): any[] | [] {
 }
 
 // returns true if ALL namespaces under a profile in default cluster are provisioned true
-export function isProfileProvisioned(namespaceSet: any[]): boolean {
+export function isProfileProvisioned(profile: any, namespaces: any[]): boolean {
   try {
-    namespaceSet.forEach((namespace: any) => {
-      // TODO: update this when we add other clusters besides default cluster
-      if (!namespace.clusters[0].provisioned) {
-        throw Error(`${namespace.name} is not provisioned yet`);
+    const { primaryClusterName } = profile;
+
+    namespaces.forEach((namespace: any) => {
+      const clusterNamespace = namespace.clusters.filter(
+        (c: any) => c.name === primaryClusterName,
```

Comment:
```suggestion
        (cluster: any) => cluster.name === primaryClusterName,
```
I recognize contextually it is obvious that c means cluster, however I wasn't sure if more explicit terminology would be beneficial.